airflow/serialization/serialized_objects.py

```diff
@@ -20,17 +20,25 @@
 import enum
 import logging
 from inspect import Parameter, signature
-from typing import Any, Dict, Iterable, List, Optional, Set, Union
+from typing import Any, Dict, Iterable, Optional, Set, Union
 
 import cattr
 import pendulum
 from dateutil import relativedelta
+
+try:
+    from functools import cache
+except ImportError:
+    from functools import lru_cache
+
+    cache = lru_cache(maxsize=None)
```

Comment:
(Style okay with this? I would have expected a new line or two after this)

Reply:
It was black-formatted, I stopped caring since we have pre-commit enabled. If black-thinks it is, good, it must be good.

Reply:
BTW. I stopped wasting my brain-cycles on formatting since we have black :)